src/workerd/jsg/modules.c++

```diff
@@ -289,8 +290,14 @@ v8::Local<v8::Module> compileEsmModule(
                           resourceColumnOffset,
                           resourceIsSharedCrossOrigin, scriptId, {},
                           resourceIsOpaque, isWasm, isModule);
-  v8::ScriptCompiler::Source source(jsg::v8Str(js.v8Isolate, content), origin);
-  return jsg::check(v8::ScriptCompiler::CompileModule(js.v8Isolate, &source));
+  if (external) {
+    v8::ScriptCompiler::Source source(
+        jsg::newExternalOneByteString(js.v8Isolate, content).ToLocalChecked(), origin);
+    return jsg::check(v8::ScriptCompiler::CompileModule(js.v8Isolate, &source));
+  } else {
+    v8::ScriptCompiler::Source source(jsg::v8Str(js.v8Isolate, content), origin);
+    return jsg::check(v8::ScriptCompiler::CompileModule(js.v8Isolate, &source));
+  }
```

Comment:
We need the branch here because module content sourced from the worker config is not immutable and we currently have to load those into the isolates heap. In theory, however, if we are creating multiple isolates with the same source, we ought to be able to benefit from using externalized strings there also... but that's a larger refactor that I don't want to do right now.